controller/controller_server/csi_controller_server.py

```diff
@@ -706,9 +706,7 @@ def _get_object_final_name(self, request, max_name_prefix_length, max_name_lengt
         return full_name[:max_name_length]
 
     def _join_object_prefix_with_name(self, prefix, name):
-        if prefix:
-            return settings.NAME_PREFIX_SEPARATOR.join((prefix, name))
-        return name
+        return settings.NAME_PREFIX_SEPARATOR.join((prefix, name))
```

Comment:
does this mean a prefix-less volume will always begin with the separator, for every storage system?

Reply:
yes, do you think its a problem?

Reply:
I am not sure if we want to begin with a separator for systems which can work without it.
@zingero what do you think?

Reply:
As far as I understand from the variable name, the separator is related to the prefix and not to the suffix. Thus, If there is nothing to separate, we don't need to use a separator

Reply:
perhaps this change lacks a context - it makes a test pass, for more info please see the ticket

Reply:
The prefix-related text in the ticket is what you wrote:
`when a user prefix is not specified, the controller should guarantee (at least for svc) that the final name doesn't start with a number.`
I'm not sure how it's related though. The prefix separator should separate the prefix and the suffix. No prefix means no separator, no?

Reply:
this change uses the separator to ensure, for all storage systems (not just svc), that the final name doesn't start with a number

Reply:
IMO, if this failure is svc-only, then we can consider adding some info in the mediator interface that would be useful only for svc.
e.g.
```
if not prefix:
    prefix = mediator.default_volume_prefix
```

Reply:
I prefer adding a default prefix when it's necessary than having a single character as an assurance for non-digit first character

Reply:
by "when it's necessary" - do you mean only for certain storage systems, as in my suggestion?
or some cross-storage default prefix (that isn't required for some storage systems)?

Reply:
As in your suggestion.

Reply:
what are suggesting for a default prefix?

Reply:
I suggest making it a configurable value and use the storage type as a default value.

Reply:
so volume will start svc_...?

Reply:
In this case, yes. Where it's not mandatory, no prefix/separator should be included.

Reply:
@oriyarde agreed?

Reply:
[here](https://github.com/IBM/ibm-block-csi-driver/pull/293#discussion_r575818570) is how I think it should be implemented, and I would use a prefix that will convey more info to the storage admin regarding the origin of this prefix, e.g. "CSI" (and not "svc")

Reply:
done, plz check

Reply:
LGTM

Reply:
@zingero plz check and resolve

Reply:
LGTM indeed

Reply:
great, @oriyarde please resolve, you opened